hashintel/hash
# Refactor Table Block to use Link Aggregations over current data

## 🌟 What is the purpose of this PR?

Earlier, the Table block was receiving its data as a part of the props being passed from the embedding application.

After the links refactoring work done in #293, this data is instead passed in via `linkedAggregations`. This PR is responsible for creating the necessary mock data, and all of the GraphQL queries needed to get this data inside of the table block.

The functionality related changes and the related links will be shared in the "What Does this Change?" section below.




## ⚠️ Known issues



- Sorting and Filtering, and changing the related entities are working, but the data is not being refetched on the query. Apollo cache keys for the `LinkedAggregation` EntityType need to be changed to update this.
- More testing needs to be done to match parity with how the table block used to previously work.
- Types need to be changed where `@todo`s are added to fix CI.

## 🚫 Blocked by




- [x] https://github.com/hashintel/hash/pull/260

## 🔍 What does this change?




- A [generic getter](https://github.com/hashintel/hash/pull/188/files#diff-66162d6d4f2e5a7fd2bfa26e899231ca9219042ba99b88b52fbdaf3b4452a8ffR66) to get a specific linked aggregation from existing linkedAggregations is created.
- Instead of the embedding application supplied data, this linked data is [used for rendering](https://github.com/hashintel/hash/pull/188/files#diff-66162d6d4f2e5a7fd2bfa26e899231ca9219042ba99b88b52fbdaf3b4452a8ffR100).
- Existing `updateEntities` operations have been updated to [update linkedAggregations](https://github.com/hashintel/hash/pull/188/files#diff-66162d6d4f2e5a7fd2bfa26e899231ca9219042ba99b88b52fbdaf3b4452a8ffR259) where needed. To support this, the object keys have been changed to match the new typing wherever needed.
- The `createLinkedAggregation` resolver has been split into 2 functions, with [`createLinkedAggregationRaw`](https://github.com/hashintel/hash/pull/188/files#diff-2ea4b166d0c7dece12dac37059bd5291ac7d4672599170268f6eb5af5176b440R11) being used to create linkedAggregations inside the [mockData](https://github.com/hashintel/hash/pull/188/files#diff-119d9d8e30c27fc2f6a023d9f971eb48a31f7541cf278bf1268a288be9a9a0acR463).
- A custom hook for [updating linked aggregations](https://github.com/hashintel/hash/pull/188/files#diff-6073056fafce7173d09afe84a7b52a768ef362a02712bcd44856fdf5e09fc397R1) in the frontend was created. [Link to mutation.](https://github.com/hashintel/hash/pull/188/files#diff-5a7290f7509993528dba17849f0b29931b871071a6a6e226013ae5f80647cb25R39)
- `linkedAggregations` is queried in [both instances](https://github.com/hashintel/hash/pull/188/files#diff-369b600ab68ad93e94a2d6e5aac7c02abb2a84cdf48674c1c96839fafe4f0cf1R72) of `page.queries.ts`.


## 🔗 Related links





- [Refactor table block to use new link fields and aggregation methods](https://app.asana.com/0/1200211978612931/1201302213469516/f)

## 🛡 What tests cover this?



No tests cover the functionality implemented as a part of this PR yet.

## ❓ How to test this?



1.  Checkout the branch. Host the table block locally and link to it from `blockPaths.json`.
1.  Run `yarn seed-data` to get the new mock data in.
1.  Make changes to the table block and confirm that things change according to the filters/sorts applied.

## 📹 Demo

No UI change was made. Functionality should still work the same.
